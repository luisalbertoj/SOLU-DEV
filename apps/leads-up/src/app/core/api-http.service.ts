import { Injectable } from '@angular/core';
// Atento a esta línea, deberás hacer que empate con tu environment
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { environment } from '../../environments/environment';

import { Observable } from 'rxjs';

const API_URL = environment.apiRoute;

@Injectable({
  providedIn: 'root',
})
export class ApiHttpService {
  protected headers;

  protected apiUrl = API_URL;

  constructor(private http: HttpClient) {
    this.headers = {
      headers: new HttpHeaders({
        'Content-Type': 'application/json',
        Accept: 'application/json',
      }),
    };
  }

  protected get(url: string): Observable<unknown> {
    return this.http.get(this.apiUrl + url, this.headers);
  }

  protected post({ url, payload }: { url: string; payload: unknown }) {
    return this.http.post('api/' + url, payload, this.headers);
  }
}
