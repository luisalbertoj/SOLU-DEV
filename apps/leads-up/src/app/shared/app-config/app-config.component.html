<a
  style="cursor: pointer"
  id="layout-config-button"
  class="layout-config-button"
  (click)="onConfigButtonClick($event)"
>
  <i class="pi pi-cog"></i>
</a>
<div
  class="layout-config"
  [ngClass]="{ 'layout-config-active': appMain.configActive$ | async }"
  (click)="appMain.onConfigClick()"
>
  <h5>Menu Type</h5>
  <div class="field-radiobutton">
    <p-radioButton
      name="layoutMode"
      value="static"
      [(ngModel)]="app.menuMode"
      inputId="layoutMode1"
    ></p-radioButton>
    <label for="layoutMode1">Static</label>
  </div>
  <div class="field-radiobutton">
    <p-radioButton
      name="layoutMode"
      value="overlay"
      [(ngModel)]="app.menuMode"
      inputId="layoutMode2"
    ></p-radioButton>
    <label for="layoutMode2">Overlay</label>
  </div>
  <div class="field-radiobutton">
    <p-radioButton
      name="layoutMode"
      value="slim"
      [(ngModel)]="app.menuMode"
      inputId="layoutMode3"
    ></p-radioButton>
    <label for="layoutMode3">Slim</label>
  </div>
  <div class="field-radiobutton">
    <p-radioButton
      name="layoutMode"
      value="horizontal"
      [(ngModel)]="app.menuMode"
      inputId="layoutMode4"
    ></p-radioButton>
    <label for="layoutMode4">Horizontal</label>
  </div>
  <div class="field-radiobutton">
    <p-radioButton
      name="layoutMode"
      value="sidebar"
      [(ngModel)]="app.menuMode"
      inputId="layoutMode5"
    ></p-radioButton>
    <label for="layoutMode5">Sidebar</label>
  </div>

  <h5>Color Scheme</h5>
  <div class="field-radiobutton">
    <p-radioButton
      name="darkMode"
      value="light"
      [(ngModel)]="app.colorScheme"
      inputId="darkMode1"
      (onClick)="changeColorScheme('light')"
    ></p-radioButton>
    <label for="darkMode1">Light</label>
  </div>
  <div class="field-radiobutton">
    <p-radioButton
      name="darkMode"
      value="dark"
      [(ngModel)]="app.colorScheme"
      inputId="darkMode2"
      (onClick)="changeColorScheme('dark')"
    ></p-radioButton>
    <label for="darkMode2">Dark</label>
  </div>

  <h5>Ripple Effect</h5>
  <p-inputSwitch
    [ngModel]="app.ripple"
    (onChange)="appMain.onRippleChange($event)"
  ></p-inputSwitch>

  <h5>Layouts</h5>
  <div class="layout-themes">
    <div *ngFor="let l of layoutColors">
      <a
        style="cursor: pointer"
        (click)="changeLayout(l.name)"
        [ngStyle]="{ 'background-color': l.color }"
      >
        <i *ngIf="app.layout === l.name" class="pi pi-check"></i>
      </a>
    </div>
  </div>

  <h5>Themes</h5>
  <div class="layout-themes">
    <div *ngFor="let t of themeColors">
      <a
        style="cursor: pointer"
        (click)="changeTheme(t.name)"
        [ngStyle]="{ 'background-color': t.color }"
      >
        <i *ngIf="app.theme === t.name" class="pi pi-check"></i>
      </a>
    </div>
  </div>
</div>
