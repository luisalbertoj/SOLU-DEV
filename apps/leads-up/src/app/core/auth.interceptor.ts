import {
  HttpEvent,
  HttpHandler,
  HttpInterceptor,
  HttpRequest,
} from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

@Injectable()
export class AuthInterceptor implements HttpInterceptor {
  intercept(
    req: HttpRequest<unknown>,
    next: HttpHandler
  ): Observable<HttpEvent<unknown>> {
    const token = localStorage.getItem('auth_token');

    if (!token) {
      return next.handle(req);
    }

    const request = req.clone({
      headers: req.headers.set('Authorization', `Bearer ${token}`),
    });

    return next.handle(request);
  }
}
