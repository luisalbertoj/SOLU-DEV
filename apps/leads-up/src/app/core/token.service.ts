import { Injectable } from '@angular/core';

@Injectable()
export class TokenProvider {
  load() {
    return new Promise((resolve) => {
      resolve(true);
    });
  }
}
