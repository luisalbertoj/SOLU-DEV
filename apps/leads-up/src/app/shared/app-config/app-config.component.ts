import { CommonModule } from '@angular/common';
import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { InputSwitchModule } from 'primeng/inputswitch';
import { RadioButtonModule } from 'primeng/radiobutton';
import { Subscription } from 'rxjs';
import { AppComponent } from '../../app.component';
import { DashboardMenuOneComponent } from '../dashboard-menu-one/dashboard-menu-one.component';
import { AppConfig } from './model/app-config';
import { ConfigService } from './services/app.config.service';

@Component({
  selector: 'solu-dev-app-config',
  standalone: true,
  imports: [CommonModule, FormsModule, RadioButtonModule, InputSwitchModule],
  providers: [ConfigService],
  templateUrl: './app-config.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class AppConfigComponent implements OnInit {
  layoutColors!: { name: string; color: string }[];

  themeColors!: { name: string; color: string }[];

  config!: AppConfig;

  subscription!: Subscription;

  constructor(
    public appMain: DashboardMenuOneComponent,
    public app: AppComponent,
    public configService: ConfigService
  ) {}

  ngOnInit() {
    this.subscription = this.configService.configUpdate$.subscribe((config) => {
      this.config = config;
    });

    this.themeColors = [
      { name: 'blue', color: '#0F8BFD' },
      { name: 'green', color: '#0BD18A' },
      { name: 'magenta', color: '#EC4DBC' },
      { name: 'orange', color: '#FD9214' },
      { name: 'purple', color: '#873EFE' },
      { name: 'red', color: '#FC6161' },
      { name: 'teal', color: '#00D0DE' },
      { name: 'yellow', color: '#EEE500' },
    ];

    this.layoutColors = [
      { name: 'blue', color: '#0F8BFD' },
      { name: 'green', color: '#0BD18A' },
      { name: 'magenta', color: '#EC4DBC' },
      { name: 'orange', color: '#FD9214' },
      { name: 'purple', color: '#873EFE' },
      { name: 'red', color: '#FC6161' },
      { name: 'teal', color: '#00D0DE' },
      { name: 'yellow', color: '#EEE500' },
    ];
  }

  changeColorScheme(scheme: string) {
    this.changeStyleSheetsColor('layout-css', 'layout-' + scheme + '.css', 1);
    this.changeStyleSheetsColor('theme-css', 'theme-' + scheme + '.css', 1);

    this.app.colorScheme = scheme;
    this.configService.updateConfig({
      ...this.config,
      ...{ dark: scheme === 'dark' },
    });
  }

  changeStyleSheetsColor(id: string, value: string, from: number) {
    const element = document.getElementById(id);
    const urlTokens = element?.getAttribute('href')?.split('/');
    if (!urlTokens) {
      return;
    }
    if (from === 1) {
      // which function invoked this function - change scheme
      urlTokens[urlTokens.length - 1] = value;
    } else if (from === 2) {
      // which function invoked this function - change color
      urlTokens[urlTokens.length - 2] = value;
    }

    const newURL = urlTokens?.join('/');

    this.replaceLink(element, newURL);
  }

  changeTheme(theme: string) {
    const themeLink: HTMLLinkElement = document.getElementById(
      'theme-css'
    ) as HTMLLinkElement;
    const href =
      'assets/theme/' + theme + '/theme-' + this.app.colorScheme + '.css';
    this.app.theme = theme;

    this.replaceLink(themeLink, href);
  }

  changeLayout(layout: string) {
    const layoutLink: HTMLLinkElement = document.getElementById(
      'layout-css'
    ) as HTMLLinkElement;
    const href =
      'assets/layout/css/' +
      layout +
      '/layout-' +
      this.app.colorScheme +
      '.css';
    this.app.layout = layout;

    this.replaceLink(layoutLink, href);
  }

  isIE() {
    return /(MSIE|Trident\/|Edge\/)/i.test(window.navigator.userAgent);
  }

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  replaceLink(linkElement: any, href: string) {
    if (this.isIE()) {
      linkElement.setAttribute('href', href);
    } else {
      const id = linkElement.getAttribute('id');
      const cloneLinkElement = linkElement.cloneNode(true);

      cloneLinkElement.setAttribute('href', href);
      cloneLinkElement.setAttribute('id', id + '-clone');

      linkElement.parentNode.insertBefore(
        cloneLinkElement,
        linkElement.nextSibling
      );

      cloneLinkElement.addEventListener('load', () => {
        linkElement.remove();
        cloneLinkElement.setAttribute('id', id);
      });
    }
  }

  onConfigButtonClick(event: Event) {
    this.appMain.configActive = !this.appMain.configActive;
    this.appMain.configActive$.next(true);
    this.appMain.configClick = true;
    event.preventDefault();
  }
}
