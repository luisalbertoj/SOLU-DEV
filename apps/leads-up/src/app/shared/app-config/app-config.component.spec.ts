import { ComponentFixture, TestBed } from '@angular/core/testing';

import { AppConfigComponent } from './app-config.component';

describe('AppConfigComponent', () => {
  let component: AppConfigComponent;
  let fixture: ComponentFixture<AppConfigComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [AppConfigComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(AppConfigComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
