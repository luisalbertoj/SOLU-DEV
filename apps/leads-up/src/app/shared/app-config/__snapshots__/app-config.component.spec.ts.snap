// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppConfigComponent this should be equal to the snapshot 1`] = `
"<a id=\\"layout-config-button\\" class=\\"layout-config-button\\" style=\\"cursor: pointer;\\"><i class=\\"pi pi-cog\\"></i></a><div class=\\"layout-config\\" ng-reflect-ng-class=\\"[object Object]\\"><h5>Menu Type</h5><div class=\\"field-radiobutton\\"><p-radiobutton name=\\"layoutMode\\" value=\\"static\\" inputid=\\"layoutMode1\\" class=\\"p-element ng-untouched ng-pristine ng-valid\\" ng-reflect-name=\\"layoutMode\\" ng-reflect-value=\\"static\\" ng-reflect-input-id=\\"layoutMode1\\" ng-reflect-model=\\"sidebar\\"><div ng-reflect-ng-class=\\"[object Object]\\" class=\\"p-radiobutton p-component\\"><div class=\\"p-hidden-accessible\\"><input type=\\"radio\\" id=\\"layoutMode1\\" name=\\"layoutMode\\" value=\\"static\\" aria-checked=\\"false\\"></div><div ng-reflect-ng-class=\\"[object Object]\\" class=\\"p-radiobutton-box\\"><span class=\\"p-radiobutton-icon\\"></span></div></div><!--bindings={}--></p-radiobutton><label for=\\"layoutMode1\\">Static</label></div><div class=\\"field-radiobutton\\"><p-radiobutton name=\\"layoutMode\\" value=\\"overlay\\" inputid=\\"layoutMode2\\" class=\\"p-element ng-untouched ng-pristine ng-valid\\" ng-reflect-name=\\"layoutMode\\" ng-reflect-value=\\"overlay\\" ng-reflect-input-id=\\"layoutMode2\\" ng-reflect-model=\\"sidebar\\"><div ng-reflect-ng-class=\\"[object Object]\\" class=\\"p-radiobutton p-component\\"><div class=\\"p-hidden-accessible\\"><input type=\\"radio\\" id=\\"layoutMode2\\" name=\\"layoutMode\\" value=\\"overlay\\" aria-checked=\\"false\\"></div><div ng-reflect-ng-class=\\"[object Object]\\" class=\\"p-radiobutton-box\\"><span class=\\"p-radiobutton-icon\\"></span></div></div><!--bindings={}--></p-radiobutton><label for=\\"layoutMode2\\">Overlay</label></div><div class=\\"field-radiobutton\\"><p-radiobutton name=\\"layoutMode\\" value=\\"slim\\" inputid=\\"layoutMode3\\" class=\\"p-element ng-untouched ng-pristine ng-valid\\" ng-reflect-name=\\"layoutMode\\" ng-reflect-value=\\"slim\\" ng-reflect-input-id=\\"layoutMode3\\" ng-reflect-model=\\"sidebar\\"><div ng-reflect-ng-class=\\"[object Object]\\" class=\\"p-radiobutton p-component\\"><div class=\\"p-hidden-accessible\\"><input type=\\"radio\\" id=\\"layoutMode3\\" name=\\"layoutMode\\" value=\\"slim\\" aria-checked=\\"false\\"></div><div ng-reflect-ng-class=\\"[object Object]\\" class=\\"p-radiobutton-box\\"><span class=\\"p-radiobutton-icon\\"></span></div></div><!--bindings={}--></p-radiobutton><label for=\\"layoutMode3\\">Slim</label></div><div class=\\"field-radiobutton\\"><p-radiobutton name=\\"layoutMode\\" value=\\"horizontal\\" inputid=\\"layoutMode4\\" class=\\"p-element ng-untouched ng-pristine ng-valid\\" ng-reflect-name=\\"layoutMode\\" ng-reflect-value=\\"horizontal\\" ng-reflect-input-id=\\"layoutMode4\\" ng-reflect-model=\\"sidebar\\"><div ng-reflect-ng-class=\\"[object Object]\\" class=\\"p-radiobutton p-component\\"><div class=\\"p-hidden-accessible\\"><input type=\\"radio\\" id=\\"layoutMode4\\" name=\\"layoutMode\\" value=\\"horizontal\\" aria-checked=\\"false\\"></div><div ng-reflect-ng-class=\\"[object Object]\\" class=\\"p-radiobutton-box\\"><span class=\\"p-radiobutton-icon\\"></span></div></div><!--bindings={}--></p-radiobutton><label for=\\"layoutMode4\\">Horizontal</label></div><div class=\\"field-radiobutton\\"><p-radiobutton name=\\"layoutMode\\" value=\\"sidebar\\" inputid=\\"layoutMode5\\" class=\\"p-element ng-untouched ng-pristine ng-valid\\" ng-reflect-name=\\"layoutMode\\" ng-reflect-value=\\"sidebar\\" ng-reflect-input-id=\\"layoutMode5\\" ng-reflect-model=\\"sidebar\\"><div ng-reflect-ng-class=\\"[object Object]\\" class=\\"p-radiobutton p-component\\"><div class=\\"p-hidden-accessible\\"><input type=\\"radio\\" id=\\"layoutMode5\\" name=\\"layoutMode\\" value=\\"sidebar\\" aria-checked=\\"false\\"></div><div ng-reflect-ng-class=\\"[object Object]\\" class=\\"p-radiobutton-box\\"><span class=\\"p-radiobutton-icon\\"></span></div></div><!--bindings={}--></p-radiobutton><label for=\\"layoutMode5\\">Sidebar</label></div><h5>Color Scheme</h5><div class=\\"field-radiobutton\\"><p-radiobutton name=\\"darkMode\\" value=\\"light\\" inputid=\\"darkMode1\\" class=\\"p-element ng-untouched ng-pristine ng-valid\\" ng-reflect-name=\\"darkMode\\" ng-reflect-value=\\"light\\" ng-reflect-input-id=\\"darkMode1\\" ng-reflect-model=\\"dark\\"><div ng-reflect-ng-class=\\"[object Object]\\" class=\\"p-radiobutton p-component\\"><div class=\\"p-hidden-accessible\\"><input type=\\"radio\\" id=\\"darkMode1\\" name=\\"darkMode\\" value=\\"light\\" aria-checked=\\"false\\"></div><div ng-reflect-ng-class=\\"[object Object]\\" class=\\"p-radiobutton-box\\"><span class=\\"p-radiobutton-icon\\"></span></div></div><!--bindings={}--></p-radiobutton><label for=\\"darkMode1\\">Light</label></div><div class=\\"field-radiobutton\\"><p-radiobutton name=\\"darkMode\\" value=\\"dark\\" inputid=\\"darkMode2\\" class=\\"p-element ng-untouched ng-pristine ng-valid\\" ng-reflect-name=\\"darkMode\\" ng-reflect-value=\\"dark\\" ng-reflect-input-id=\\"darkMode2\\" ng-reflect-model=\\"dark\\"><div ng-reflect-ng-class=\\"[object Object]\\" class=\\"p-radiobutton p-component\\"><div class=\\"p-hidden-accessible\\"><input type=\\"radio\\" id=\\"darkMode2\\" name=\\"darkMode\\" value=\\"dark\\" aria-checked=\\"false\\"></div><div ng-reflect-ng-class=\\"[object Object]\\" class=\\"p-radiobutton-box\\"><span class=\\"p-radiobutton-icon\\"></span></div></div><!--bindings={}--></p-radiobutton><label for=\\"darkMode2\\">Dark</label></div><h5>Ripple Effect</h5><p-inputswitch class=\\"p-element ng-untouched ng-pristine ng-valid\\" ng-reflect-model=\\"true\\"><div ng-reflect-ng-class=\\"[object Object]\\" class=\\"p-inputswitch p-component\\"><div class=\\"p-hidden-accessible\\"><input type=\\"checkbox\\" role=\\"switch\\" aria-checked=\\"false\\"></div><span class=\\"p-inputswitch-slider\\"></span></div></p-inputswitch><h5>Layouts</h5><div class=\\"layout-themes\\"><div><a style=\\"cursor: pointer; background-color: rgb(15, 139, 253);\\" ng-reflect-ng-style=\\"[object Object]\\"><i class=\\"pi pi-check\\"></i><!--bindings={
  \\"ng-reflect-ng-if\\": \\"true\\"
}--></a></div><div><a style=\\"cursor: pointer; background-color: rgb(11, 209, 138);\\" ng-reflect-ng-style=\\"[object Object]\\"><!--bindings={
  \\"ng-reflect-ng-if\\": \\"false\\"
}--></a></div><div><a style=\\"cursor: pointer; background-color: rgb(236, 77, 188);\\" ng-reflect-ng-style=\\"[object Object]\\"><!--bindings={
  \\"ng-reflect-ng-if\\": \\"false\\"
}--></a></div><div><a style=\\"cursor: pointer; background-color: rgb(253, 146, 20);\\" ng-reflect-ng-style=\\"[object Object]\\"><!--bindings={
  \\"ng-reflect-ng-if\\": \\"false\\"
}--></a></div><div><a style=\\"cursor: pointer; background-color: rgb(135, 62, 254);\\" ng-reflect-ng-style=\\"[object Object]\\"><!--bindings={
  \\"ng-reflect-ng-if\\": \\"false\\"
}--></a></div><div><a style=\\"cursor: pointer; background-color: rgb(252, 97, 97);\\" ng-reflect-ng-style=\\"[object Object]\\"><!--bindings={
  \\"ng-reflect-ng-if\\": \\"false\\"
}--></a></div><div><a style=\\"cursor: pointer; background-color: rgb(0, 208, 222);\\" ng-reflect-ng-style=\\"[object Object]\\"><!--bindings={
  \\"ng-reflect-ng-if\\": \\"false\\"
}--></a></div><div><a style=\\"cursor: pointer; background-color: rgb(238, 229, 0);\\" ng-reflect-ng-style=\\"[object Object]\\"><!--bindings={
  \\"ng-reflect-ng-if\\": \\"false\\"
}--></a></div><!--bindings={
  \\"ng-reflect-ng-for-of\\": \\"[object Object],[object Object\\"
}--></div><h5>Themes</h5><div class=\\"layout-themes\\"><div><a style=\\"cursor: pointer; background-color: rgb(15, 139, 253);\\" ng-reflect-ng-style=\\"[object Object]\\"><i class=\\"pi pi-check\\"></i><!--bindings={
  \\"ng-reflect-ng-if\\": \\"true\\"
}--></a></div><div><a style=\\"cursor: pointer; background-color: rgb(11, 209, 138);\\" ng-reflect-ng-style=\\"[object Object]\\"><!--bindings={
  \\"ng-reflect-ng-if\\": \\"false\\"
}--></a></div><div><a style=\\"cursor: pointer; background-color: rgb(236, 77, 188);\\" ng-reflect-ng-style=\\"[object Object]\\"><!--bindings={
  \\"ng-reflect-ng-if\\": \\"false\\"
}--></a></div><div><a style=\\"cursor: pointer; background-color: rgb(253, 146, 20);\\" ng-reflect-ng-style=\\"[object Object]\\"><!--bindings={
  \\"ng-reflect-ng-if\\": \\"false\\"
}--></a></div><div><a style=\\"cursor: pointer; background-color: rgb(135, 62, 254);\\" ng-reflect-ng-style=\\"[object Object]\\"><!--bindings={
  \\"ng-reflect-ng-if\\": \\"false\\"
}--></a></div><div><a style=\\"cursor: pointer; background-color: rgb(252, 97, 97);\\" ng-reflect-ng-style=\\"[object Object]\\"><!--bindings={
  \\"ng-reflect-ng-if\\": \\"false\\"
}--></a></div><div><a style=\\"cursor: pointer; background-color: rgb(0, 208, 222);\\" ng-reflect-ng-style=\\"[object Object]\\"><!--bindings={
  \\"ng-reflect-ng-if\\": \\"false\\"
}--></a></div><div><a style=\\"cursor: pointer; background-color: rgb(238, 229, 0);\\" ng-reflect-ng-style=\\"[object Object]\\"><!--bindings={
  \\"ng-reflect-ng-if\\": \\"false\\"
}--></a></div><!--bindings={
  \\"ng-reflect-ng-for-of\\": \\"[object Object],[object Object\\"
}--></div></div>"
`;
